security/advancedtls/advancedtls.go

```diff
@@ -35,6 +35,7 @@ import (
 
 // VerificationFuncParams contains the parameters available to users when implementing CustomVerificationFunc.
 type VerificationFuncParams struct {
+	// This field is only meaningful for client side. On server side, this field would be an empty string.
```

Comment:
Can we have two structures one for client and one for server parameters? This way we won't have to rely on the user to do the right think and not set this field on the server-side.

If not, it would be worth mentioning, in this comment, what happens if servers set this field, does the application crash, does the connection fail, etc?

Reply:
We can split this field into two, but then we have to split `CustomVerificationFunc`, and then  `advancedTLSCreds` needs to take both `ClientCustomVerificationFunc` and `ServerCustomVerificationFunc`... This might make code a bit more complex and I am not sure if we want to change it.
All the fields in  `VerificationFuncParams` are read-only, meaning users can read these fields when making their custom verifications, but setting them wouldn't have any effect. I added a comment above the line and hope it could resolve the confusion I caused here.